Case insensitive date_trunc, date_part; toStartOfSeconds needs DateTi…

Peter-Sh opened the pull request:
This PR
* makes comparison of time units (SECONDS, MINUTES etc.) in functions date_trunc, date_part case insensitive (based on #52)
* add toDateTime64(col, 1) conversion when using seconds unit as it is required type for toStartOfSeconds clickhouse function


this bug make it can not work with tableau filter date range
and cannot work with google datastudio filter

ildus commented:
Thanks for PR. Could you add some tests? It would help me to understand what's happening there. And would be good to follow same formatting rules as the other code around.

Peter-Sh commented:
Ok, I thought about tests, but did not notice that they exists %) I'll try to make test from my test queries.

Can you explain about formatting rules?

I tried to follow "code around", but may be have missed something. I've added curly brackets in if .. } else if { .. else if { sequence because there are two lines of code now in "second" condition and I though it should be all with braces or all without braces, but not mixed.

Peter-Sh commented:
I see, I missed new lines before "{", will fix.

Peter-Sh commented:
Hi, @ildus!

I have updated tests to reflect changes in this PR (case insensitivity and type cast to DateTime64 for toStartOfSecond) and fixed code formatting.
Additionally I've made two little updates because some other tests are failing with recent versions of clickhouse, now all tests in CI should pass.

ildus commented:
Thanks!
